P2022/tests/header/3way.hpp

```diff
@@ -1,16 +1,32 @@
 
 
 #include <iterator>
+#include <concepts>
 
 namespace std::ranges
 {
+    template<
+        input_iterator I1,
+        input_iterator I2,
+        class Comp,
+        class Proj1,
+        class Proj2
+    >
+    using lexicographical_compare_three_way_result_t =
+        invoke_result_t<Comp, invoke_result_t<Proj1, I1>, invoke_result_t<Proj2, I2>>;
```

Comment:
if replacing `invoke_reasult_t<Proj1, I1>` with `projected<I1, Proj1>` (and the same for `2`) - t1.cpp does not compile.

Reply:
can you please add the error message, or merge this so i could pull it ?

Reply:
invoke_result_t<Proj1, I1> and std::projected<Proj1, I1> are two different objects, so the comparator probably doesnt expect projected as a type. 
the comparator expects comparable types but std::projected is not.

Reply:
as i suspected 
'three_way_comparable_with<std::projected<__gnu_cxx::__normal_iterator<int *, std::vector<int> >, std::identity>, std::projected<__gnu_cxx::__normal_iterator<int *, std::vector<int> >, std::identity> >' evaluated to false

Reply:
you may use this but its ugly as hell and i think the invoke version is better 
invoke_result_t<Comp, typename projected<I1, Proj1>::value_type , typename projected<I2, Proj2>::value_type>;